ATP Compile Issues (

Working on Windows Subsystem for Linux/Ubuntu 18.04 LTS, on make getting the error
test.cc:14:10: fatal error: cppunit/ui/text/TestRunner.h: No such file or directory
```
logger.hh:234:6: error: mangled name for ‘void TrafficProfiles::Logger::_log(TrafficProfiles::Logger::Level, T, Tail&& ...) [with T = long int (*)(long int*) throw (); Tail = {const char (&)[6], const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]’ will change in C++17 because the exception specification is part of a function type [-Werror=noexcept-type]
 void Logger::_log(const Level lvl,T head, Tail&&... tail){
      ^~~~~~
logger.hh:234:6: error: mangled name for ‘void TrafficProfiles::Logger::_log(TrafficProfiles::Logger::Level, T, Tail&& ...) [with T = const char*; Tail = {long int (&)(long int*) throw (), const char (&)[6], const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]’ will change in C++17 because the exception specification is part of a function type [-Werror=noexcept-type]
logger.hh:234:6: error: mangled name for ‘void TrafficProfiles::Logger::_log(TrafficProfiles::Logger::Level, T, Tail&& ...) [with T = std::__cxx11::basic_string<char>; Tail = {const char (&)[29], long int (&)(long int*) throw (), const char (&)[6], const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]’ will change in C++17 because the exception specification is part of a function type [-Werror=noexcept-type]
logger.hh:234:6: error: mangled name for ‘void TrafficProfiles::Logger::_log(TrafficProfiles::Logger::Level, T, Tail&& ...) [with T = const char*; Tail = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const char (&)[29], long int (&)(long int*) throw (), const char (&)[6], const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]’ will change in C++17 because the exception specification is part of a function type [-Werror=noexcept-type]
logger.hh:234:6: error: mangled name for ‘void TrafficProfiles::Logger::_log(TrafficProfiles::Logger::Level, T, Tail&& ...) [with T = std::__cxx11::basic_string<char>; Tail = {const char (&)[11], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const char (&)[29], long int (&)(long int*) throw (), const char (&)[6], const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]’ will change in C++17 because the exception specification is part of a function type [-Werror=noexcept-type]
logger.hh:234:6: error: mangled name for ‘void TrafficProfiles::Logger::_log(TrafficProfiles::Logger::Level, T, Tail&& ...) [with T = const char*; Tail = {std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, const char (&)[11], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const char (&)[29], long int (&)(long int*) throw (), const char (&)[6], const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]’ will change in C++17 because the exception specification is part of a function type [-Werror=noexcept-type]
logger.hh:206:6: error: mangled name for ‘void TrafficProfiles::Logger::log(TrafficProfiles::Logger::Level, Tail&& ...) [with Tail = {const char (&)[32], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, const char (&)[11], std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, const char (&)[29], long int (&)(long int*) throw (), const char (&)[6], const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >}]’ will change in C++17 because the exception specification is part of a function type [-Werror=noexcept-type]
 void Logger::log(const Level lvl, Tail&&... args) {
      ^~~~~~
cc1plus: all warnings being treated as errors
Makefile:78: recipe for target 'traffic_profile_master.o' failed
make: *** [traffic_profile_master.o] Error 1
make: *** Waiting for unfinished jobs....
```



Gem5 Compilation will also fail.
Workarounds so far:
- For standalone compilation, change the Makefile Line 21 to use -std=c++17
`CXX_FLAGS       := $(PROTOBUF_C_FLAGS) -std=c++17 -Wall -Werror -Wextra -Wno-unused-parameter -Wno-unused-variable `
- For gem5 compilation disablling Werror will help, but is not best practice. Delete from Sconstruct file in line line 377 `'-Werror',`
Adding `'-Wno-error=noexcept-type'` caused python issues

--- a/traffic_profile_master.cc
+++ b/traffic_profile_master.cc
@@ -300,7 +300,7 @@ bool TrafficProfileMaster::receive(
             if (0 == ot) {
             ERROR("TrafficProfileMaster::receive [", this->name,
                     "] address ", Utilities::toHex(packet->addr()),
-                    "negative OT detected at time", time, "stats",
+                    "negative OT detected at time", t, "stats",
                     stats.dump());
             }
             ot--;

